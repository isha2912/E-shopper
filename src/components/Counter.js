import React from 'react';
import PropTypes from 'prop-types';
import './Counter.css';

function Counter(props) {
  const { onIncrement, onDecrement, value } = props;
  return (
    <div className="counter">
      <button type="button" className="button-decrement" onClick={onDecrement}>
        -
      </button>
      <div className="in-basket">
        {value}
        {' '}
        in Basket
      </div>
      <button type="button" className="button-increment" onClick={onIncrement}>
        +
      </button>
    </div>
  );
}

export default Counter;

Counter.propTypes = {
  value: PropTypes.number.isRequired,
  onDecrement: PropTypes.func.isRequired,
  onIncrement: PropTypes.func.isRequired,
};
