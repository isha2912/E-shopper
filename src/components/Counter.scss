@import "../styles/index.scss";
.counter {
  display: flex;
  justify-content: space-around;
  background-color: wheat;
  width: 100%;
  .in-basket {
    flex-grow: 2;
  }
}
