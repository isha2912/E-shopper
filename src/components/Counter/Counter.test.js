/* eslint-disable react/jsx-props-no-spreading */
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Counter from './Counter';

describe(Counter.name, () => {
  const mockProp = {
    value: 0,
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('should display the current count', () => {
    render(<Counter {...mockProp} />);

    expect(screen.getByText('+'));
    expect(screen.getByText('0 in Basket'));
    expect(screen.getByText('-'));
  });
  test('should match snapshot', () => {
    const { container } = render(<Counter {...mockProp} />);

    expect(container).toMatchSnapshot();
  });

  test('Should increment value onIncrement', () => {
    render(<Counter {...mockProp} />);

    const incrementCount = screen.getByText('+');
    expect(incrementCount.tagName).toBe('BUTTON');
  });
  test('Should decrement value onDecrement', () => {
    render(<Counter {...mockProp} />);

    const incrementCount = screen.getByText('-');
    expect(incrementCount.tagName).toBe('BUTTON');
  });

  test('Should decrement value onDecrement', () => {
    render(<Counter {...mockProp} />);

    const decrementCount = screen.getByText('-');
    fireEvent.click(decrementCount);
    expect(mockProp.onDecrement).toHaveBeenCalled();
  });

  test('Should Increment value onIncrement', () => {
    render(<Counter {...mockProp} />);

    const incrementCount = screen.getByText('+');
    fireEvent.click(incrementCount);
    expect(mockProp.onIncrement).toHaveBeenCalled();
  });
});
