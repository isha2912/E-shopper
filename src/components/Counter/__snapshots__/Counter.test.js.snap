// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match snapshot 1`] = `
<div>
  <div
    class="counter"
  >
    <button
      class="button-decrement"
      type="button"
    >
      -
    </button>
    <div
      class="in-basket"
    >
      0
       
      in Basket
    </div>
    <button
      class="button-increment"
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
